src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,157 @@
+import classNames from 'classnames';
+import React, { useEffect, useState, useMemo } from 'react';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+import TodoTitleField from '../TodoTitleField/TodoTitleFiled';
+
+type Props = {
+  todo: Todo;
+  onDeleteTodo: (todoId: number) => void;
+  deletingCompleted: boolean;
+  onToggleStatus: () => void;
+  isChanging: boolean;
+  currentlyChanging: Filter;
+  onTitleChange: (todo: Todo, value: string) => void;
+};
+
+const TodoComponent: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  deletingCompleted,
+  onToggleStatus,
+  isChanging,
+  currentlyChanging,
+  onTitleChange,
+}) => {
+  const [hasSpinner, setHasSpinner] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const activeBecauseDeletingCompleted = todo.completed && deletingCompleted;
+  const isActive = useMemo(
+    () => {
+      let isActiveTemp = activeBecauseDeletingCompleted || hasSpinner;
+
+      if (currentlyChanging) {
+        if (todo.completed && currentlyChanging === Filter.COMPLETED) {
+          isActiveTemp = true;
+        }
+
+        if (!todo.completed && currentlyChanging === Filter.ACTIVE) {
+          isActiveTemp = true;
+        }
+      }
+
+      return isActiveTemp;
+    }, [
+      activeBecauseDeletingCompleted, hasSpinner,
+    ],
+  );
+
+  useEffect(() => {
+    if (!isChanging) {
+      setHasSpinner(false);
+    }
+  }, [isChanging]);
+
+  const deleteHandler = (todoId:number) => {
+    onDeleteTodo(todoId);
+    setHasSpinner(true);
+  };
+
+  const toggleHandler = () => {
+    setHasSpinner(true);
+    onToggleStatus();
+  };
+
+  const showEditForm = () => {
+    setIsEditing(true);
+  };
+
+  useEffect(() => {
+    const listener = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        setIsEditing(false);
+      }
+    };
+
+    document.addEventListener('keydown', listener);
```

Comment:
actually, in React you should use `onKeyDown` prop

Reply:
i will stick to this variant because here i dont have a document component to attche onKeyDown to